### PR TITLE
Set/Get ItemStack from ThrownPotion. Adds BUKKIT-3197

### PR DESCRIPTION
**Pull request by oloflarsson**
Corresponding CraftBukkit PR: https://github.com/Bukkit/CraftBukkit/pull/958
Leaky Ticket: https://bukkit.atlassian.net/browse/BUKKIT-3197

### CODE REVIEW

**Comment by natemort** [comment]
You'll need a Leaky ticket for this to get anywhere.

**Comment by oloflarsson** [comment]
Really? Why?

**Comment by Cogito** [comment]
@oloflarsson In short, because we do Issue Driven Development. Every code change is backed by a ticket, and the tickets automatically gather all changes that relate to it in the one place. This way it becomes easy to co-ordinate changes for the same issue, and make sure we have followed practices for every code commit.

It also help generate change logs and things like that.

**Comment by oloflarsson** [comment]
https://bukkit.atlassian.net/browse/BUKKIT-3197

**Comment by shinyquagsire23** [comment]
Now just append something like <b> - Fixes BUKKIT-3197</b> to the title and commit.

**Comment by oloflarsson** [comment]
@shinyquagsire23 Excellent suggestion :) Thanks! Just did so.

File: src/main/java/org/bukkit/entity/ThrownPotion.java [hunk 1]
```diff
@@ -13,4 +14,21 @@
      * @return The potion effects
      */
     public Collection<PotionEffect> getEffects();
+
```
Comment (gmcferrin) [inline src/main/java/org/bukkit/entity/ThrownPotion.java]: Spaces are bad.

File: src/main/java/org/bukkit/entity/ThrownPotion.java [hunk 2]
```diff
@@ -13,4 +14,21 @@
      * @return The potion effects
      */
     public Collection<PotionEffect> getEffects();
+    
+    /**
+     * Returns a copy of the item stack for this thrown potion.
+     * Altering this copy will not alter the thrown potion directly.
+     * If you want to alter the thrown potion you must use the {@link #setItemStack(ItemStack) setItemStack} method.
+     *
```
Comment (gmcferrin) [inline src/main/java/org/bukkit/entity/ThrownPotion.java:22]: Spaces are bad.

File: src/main/java/org/bukkit/entity/ThrownPotion.java [hunk 3]
```diff
@@ -13,4 +14,21 @@
...
+    
+    /**
+     * Returns a copy of the item stack for this thrown potion.
+     * Altering this copy will not alter the thrown potion directly.
+     * If you want to alter the thrown potion you must use the {@link #setItemStack(ItemStack) setItemStack} method.
+     * 
+     * @return A copy of the item stack for this thrown potion.
+     */
+    public ItemStack getItemStack();
+
```
Comment (gmcferrin) [inline src/main/java/org/bukkit/entity/ThrownPotion.java]: Spaces are bad.

File: src/main/java/org/bukkit/entity/ThrownPotion.java [hunk 4]
```diff
@@ -13,4 +14,21 @@
...
+     * If you want to alter the thrown potion you must use the {@link #setItemStack(ItemStack) setItemStack} method.
+     * 
+     * @return A copy of the item stack for this thrown potion.
+     */
+    public ItemStack getItemStack();
+    
+    /**
+     * Set the item stack for this thrown potion.
+     * The ItemStack must be a potion. Otherwise an exception is thrown.
+     *
```
Comment (gmcferrin) [inline src/main/java/org/bukkit/entity/ThrownPotion.java:30]: Spaces are bad.

**Comment by oloflarsson** [comment]
@gmcferrin Thank you, now fixed.

**Comment by gmcferrin** [comment]
@oloflarsson Still have one space that needs fixing.

**Comment by gmcferrin** [comment]
Pulled.

**Pull request closed without merging**